IDE/Renesas/e2studio/Projects/test/src/wolf_client.c

```diff
@@ -0,0 +1,201 @@
+/* wolf_client.c
+ *
+ * Copyright (C) 2006-2019 wolfSSL Inc.
+ *
+ * This file is part of wolfSSL.
+ *
+ * wolfSSL is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * wolfSSL is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1335, USA
+ */
+
+#include <stdio.h>
+#include <string.h>
+#include "r_t4_itcpip.h"
+
+#include "wolfssl/wolfcrypt/settings.h"
+#include "wolfssl/ssl.h"
+#include "wolfssl/certs_test.h"
+#include "key_data.h"
+
+#define SIMPLE_TLSSEVER_IP       "192.168.1.45"
+#define SIMPLE_TLSSERVER_PORT    "11111"
+
+ER    t4_tcp_callback(ID cepid, FN fncd , VP p_parblk);
+uint32_t g_encrypted_root_public_key[140];
+
+static WOLFSSL_CTX *client_ctx;
+
+static int my_IORecv(WOLFSSL* ssl, char* buff, int sz, void* ctx)
+{
+    int ret;
+    ID  cepid;
+
+    if(ctx != NULL)
+        cepid = *(ID *)ctx;
+    else
+        return WOLFSSL_CBIO_ERR_GENERAL;
+
+    ret = tcp_rcv_dat(cepid, buff, sz, TMO_FEVR);
+    if(ret > 0)return ret;
+    else         return WOLFSSL_CBIO_ERR_GENERAL;
+}
+
+static int my_IOSend(WOLFSSL* ssl, char* buff, int sz, void* ctx)
+{
+    int ret;
+    ID  cepid;
+
+    if(ctx != NULL)
+    	cepid = *(ID *)ctx;
+    else
+    	return WOLFSSL_CBIO_ERR_GENERAL;
+
+    ret = tcp_snd_dat(cepid, buff, sz, TMO_FEVR);
+    if(ret == sz)return ret;
```

Comment:
Please adjust if/else formatting to match multi-line style for readability here, thanks.

Reply:
Done